### PR TITLE
Apply branch protection rules

### PR DESCRIPTION
**Pull request by stephanme**
- first experiments

### CODE REVIEW

**Comment by beyhan** [comment]
I have asked [here](https://kubernetes.slack.com/archives/CCK68P2Q2/p1668522516694529) the [sig-k8s-infra](https://kubernetes.slack.com/messages/sig-k8s-infra) WG questions we have with Stephan about the `branchprotector` tool.

**Comment by stephanme** [comment]
Sure, we will announce the rollout of branch protection automation. I guess it will take some time until we have a working implementation as the branchprotector tool misses some options.

**Comment by beyhan** [comment]
I didn't get a concrete answer by the `prow` community but looking into github APIs and the `prow` implementation I think that we need to add the `required_approving_review_count` parameter as documented in the [Github API docs](https://docs.github.com/en/rest/branches/branch-protection?apiVersion=2022-11-28#update-pull-request-review-protection) to the prow code [here](https://github.com/kubernetes/test-infra/blob/02c787924618a7f1aa8b1a4af92c43f9712fa6d4/prow/cmd/branchprotector/request.go#L116-L134)

**Comment by beyhan** [comment]
I create a [PR](https://github.com/kubernetes/test-infra/pull/28263) for the branchprotector to add the missing feature for PR bypass configuration.

**Comment by beyhan** [comment]
The [PR](https://github.com/kubernetes/test-infra/pull/28263) for the branchprotector is merged. I tried the feature and it looks good. We should be able now to implement the [branchprotection RFC](https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0015-branch-protection.md).

File: org/readme.md [hunk 1]
```diff
@@ -28,6 +29,44 @@ Organization admins are:
...
+        # automation generates config for all repos belonging to a WG unless an explicit configuration exists in branchprotection.yml
+        <projectname>:
+          protect: true
+          enforce_admins: true
+          allow_force_pushes: false
+          allow_deletions: false
+          allow_disabled_policies: true  # needed to allow branches w/o branch protection
+          required_pull_request_reviews:
+            dismiss_stale_reviews: true
+            require_code_owner_reviews: true
```
Comment (rkoster) [inline org/readme.md:59]: What happens if no code owners have been configured for a repo?

Comment (stephanme) [reply org/readme.md:59]: https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/managing-a-branch-protection-rule

> Optionally, to require review from a code owner when the pull request affects code that has a designated owner, select **Require review from Code Owners**.

This flag has only affect, if code owners have been configured.

Tested with https://github.com/cloudfoundry/branchprotection-test/pull/1

**Review by rkoster — APPROVED** [review]

**Review by emalm — APPROVED** [review]

**Review by AP-Hunt — APPROVED** [review]

**Review by beyhan — APPROVED** [review]

**Comment by emalm** [comment]
For the record, since @Gerg asked about it: each WG would opt into these branch-protection rules with a new `config.generate_rfc0015_branch_protection_rules` parameter in the embedded YAML document in the charter file. (https://github.com/cloudfoundry/community/blob/c62f00f8dbab5301f45c62d2b290357bf0537f32/org/org_management.py#L117 is the place where the automation checks for that new flag.)

**Pull request merged**